broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/NewPartitionTransitionImpl.java

```diff
@@ -71,67 +73,48 @@ public void updateTransitionContext(final PartitionTransitionContext transitionC
     steps.forEach(step -> step.onNewRaftRole(context, role));
 
     final ActorFuture<Void> nextTransitionFuture = concurrencyControl.createFuture();
+    final var nextTransition =
+        new PartitionTransitionProcess(steps, concurrencyControl, context, term, role);
+    nextTransitionFuture.onComplete((v, e) -> lastTransition = nextTransition);
+
+    final var ongoingTransitionFuture =
+        currentTransitionFuture == null
+            ? concurrencyControl.createCompletedFuture()
+            : currentTransitionFuture;
+
+    // For safety reasons we have to immediately replace the current transition future with the next
+    // transition future, such that we make sure that we enqueue all transitions after another.
+    //
+    // This means we will always add the next transition to the tail, this can become a chain of
+    // futures if we have many transitions after another. Ideally we would execute after the current
+    // transition only the last enqueue one. This should be implemented as soon we have time for it.
+    currentTransitionFuture = nextTransitionFuture;
 
-    concurrencyControl.run(
-        () -> {
-          if (currentTransition != null) {
-            LOG.info(
-                format(
-                    "Transition to %s on term %d requested while another transition is still running",
-                    role, term));
-            currentTransition.cancel(); // this will drop any subsequent transition steps
-
-            // schedule new transition as soon as the current step of the current transition
-            // has completed
-            concurrencyControl.runOnCompletion(
-                currentTransitionFuture,
-                (ok, error) -> cleanupLastTransition(nextTransitionFuture, term, role));
+    ongoingTransitionFuture.onComplete(
+        (nothing, error) -> {
```

Comment:
:question: Dumb question, but do we not care if there's an error or not? I know we check it below, but I assumed that was a typo and you meant to check `e` in the other callback. If not, then - do we not care about `e` in that other callback? :smile:

Reply:
When error, the following transitions should not be executed.

Reply:
Why not?

Reply:
Ah.. sorry. I was not thinking clearly. 
> When error, the following transitions should not be executed.

I withdraw this comment :slightly_smiling_face: